stackrox/stackrox
# ROX-10982: Increase z-index so Search modal overlaps banners in UI

## Description

### Problem

Certificate expiration banners cause masthead to overlap search filter input:
![Banners-2-overlap-Search](https://user-images.githubusercontent.com/11862657/182258969-46318035-5d57-430e-b4c1-c3c3cb99a20c.png)

### Analysis

1. MainPage.tsx can conditionally render any of the following as `Banner` element with `isSticky` prop:
    * `UnreachableWarning`
    * `CredentialExpiryBanner` for central
    * `CredentialExpiryBanner` for scanner
    * `VersionOutOfDate`

2. `pf-c-banner--m-sticky-ZIndex` variable has value `300` see https://www.patternfly.org/v4/components/banner#css-variables

3. Masthead.tsx renders `PageHeader` element for which `--pf-c-page__header--ZIndex` variable also has value `300` see https://www.patternfly.org/v4/components/page#css-variables

4. `SearchModalContainer` renders `div` which has `z-sm-200` in `classNames` prop

5. Style rule (in app.tw.css file) specifies top edge of Search modal without regard to banners:

    ```css
    .search-modal {
        position: absolute;
        left: 0;
        top: 76px;
        height: calc(100% - 76px);
    }
    ```

### Changed files

1. Edit MainPage.tsx file

2. Move VersionOutOfDate.tsx file from Containers to Containers/MainPage folder:
    * In parallel to CredentialExpiryBanner.tsx file, because only MainPage.tsx file imports them.
    * In contrast to VersionOutOfDate.tsx file, because both Login/LoginPage.tsx and MainPage/MainPage.tsx import it.

3. Edit SearchModal.tsx file
    * Replace `z-sm-200` with `z-md-300` from trumps.css file

## Checklist

## Testing Performed

### manual testing

1. Open Search modal in ordinary scenario with no banners to verify no regression
    ![Banners-0](https://user-images.githubusercontent.com/11862657/182258998-f3f3877b-6c21-4cf9-9bda-9cb83335efb7.png)

2. Temporarily change code to render both certificate expiration banners as reported

    * Incorrect before the change: Search overlapped by masthead (see above)

    * Correct after the change: Search overlaps masthead
        ![Banners-2-overlapped-by-Search](https://user-images.githubusercontent.com/11862657/182259063-7e4b0df1-a8a6-41a8-bcba-01e12dd1ed0b.png)

3. Temporarily change code to render all 4 banners as worst case

    * Correct after the change: Search overlaps banners
        ![Banners-4-overlapped-by-Search](https://user-images.githubusercontent.com/11862657/182259083-29b7a634-5c0d-4f51-bef0-912ea0119285.png)

### integration testing

* search.test.js
